suite/suite.go

```diff
@@ -21,24 +22,31 @@ var matchMethod = flag.String("testify.m", "", "regular expression to select tes
 // retrieving the current *testing.T context.
 type Suite struct {
 	*assert.Assertions
+	mu      sync.Mutex
 	require *require.Assertions
 	t       *testing.T
 }
 
 // T retrieves the current *testing.T context.
 func (suite *Suite) T() *testing.T {
+	suite.mu.Lock()
```

Comment:
This is a read-only operation; I'm not sure it needs a mutex?

Reply:
Maybe a `sync.RWMutex` and `suite.mu.RLock()`

Reply:
Yes, I think it is a ```sync.RWMutex```.